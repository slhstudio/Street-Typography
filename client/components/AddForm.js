import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'; 

const StyledForm = styled.form.attrs({
  className: 'form'
})` 
  display: flex;
  flex-direction: column;
  margin: 5%;
`;

class AddForm extends Component {        
  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
  }

  doSubmit = (e) => {
    this.props.onSubmit(e);
  }

  render () {
    return (
      <StyledForm onSubmit={this.doSubmit}>   
        {this.props.children}
      </StyledForm> 
    )
  }
}
             

export default AddForm;