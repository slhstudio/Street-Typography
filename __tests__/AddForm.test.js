import React from 'react';
import AddForm from '../client/components/AddForm';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeProps = {
  onSubmit: jest.fn(),
  children: ['child1', 'child2', 'child3']
}

describe('<AddForm/>', () => {
  it ('renders and displays properly', () => {
    const wrapper = shallow(<AddForm {...fakeProps}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});