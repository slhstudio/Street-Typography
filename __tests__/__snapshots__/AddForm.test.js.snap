// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddForm/> renders and displays properly 1`] = `
<styled.form
  onSubmit={[Function]}
>
  child1
  child2
  child3
</styled.form>
`;
